src/components/svg-canvas-graph.tsx

```diff
@@ -160,54 +165,85 @@ const SvgCanvas = () => {
                 if (isNodeSupportSnapLine(node, graph.current)) {
                     // previous move operation may have active polylines
                     // use and check if they are still valid, if not, remove and recalculate
-                    let nowPolylines = activeSnapLines;
+                    let nowSnapLines = activeSnapLines;
+                    let nowSnapPoints = activeSnapPoints;
 
                     // check if cursor left the polyline and remove it from active polylines
-                    if (nowPolylines.length !== 0) {
-                        nowPolylines = nowPolylines.filter(l => getSnapLineDistance(l, toX, toY) <= 6);
+                    if (nowSnapLines.length !== 0) {
+                        nowSnapLines = nowSnapLines.filter(l => getSnapLineDistance(l, toX, toY) <= 6);
+                    }
+
+                    if (nowSnapPoints) {
+                        if (nowSnapLines.length === 0) {
+                            nowSnapPoints = undefined;
+                        } else if (Math.hypot(toX - nowSnapPoints.x, toY - nowSnapPoints.y) > 6) {
+                            nowSnapPoints = undefined;
+                        }
+                    }
+
+                    if (!nowSnapPoints && nowSnapLines.length === 1) {
+                        const l = nowSnapLines[0];
+                        const { snapPoint, distance } = getNearestSnapPoints(
+                            graph.current,
+                            nodesInViewRange.filter(node => !selected.has(node)),
+                            toX,
+                            toY,
+                            l
+                        );
+                        console.log(snapPoint, distance);
+                        if (distance <= 3) {
```

Comment:
What does this condition mean and what we should do next? Some comments should be added.